packages/excalidraw/components/Stats/MultiDimension.tsx

```diff
@@ -0,0 +1,337 @@
+      : {}),
+  };
+};
+
+const resizeElementInGroup = (
+  anchorX: number,
+  anchorY: number,
+  property: MultiDimensionProps["property"],
+  scale: number,
+  latestElement: ExcalidrawElement,
+  origElement: ExcalidrawElement,
+  elementsMap: ElementsMap,
+  originalElementsMap: ElementsMap,
+) => {
+  const updates = getResizedUpdates(anchorX, anchorY, scale, origElement);
+
+  mutateElement(latestElement, updates, false);
+  const boundTextElement = getBoundTextElement(
+    origElement,
+    originalElementsMap,
+  );
+  if (boundTextElement) {
+    const newFontSize = boundTextElement.fontSize * scale;
+    updateBoundElements(latestElement, elementsMap, {
+      newSize: { width: updates.width, height: updates.height },
+    });
+    const latestBoundTextElement = elementsMap.get(boundTextElement.id);
+    if (latestBoundTextElement && isTextElement(latestBoundTextElement)) {
+      mutateElement(
+        latestBoundTextElement,
+        {
+          fontSize: newFontSize,
+        },
+        false,
+      );
+      handleBindTextResize(
+        latestElement,
+        elementsMap,
+        property === "width" ? "e" : "s",
+        true,
+      );
+    }
+  }
+};
+
+const resizeGroup = (
+  nextWidth: number,
+  nextHeight: number,
+  initialHeight: number,
+  aspectRatio: number,
+  anchor: Point,
+  property: MultiDimensionProps["property"],
+  latestElements: ExcalidrawElement[],
+  originalElements: ExcalidrawElement[],
+  elementsMap: ElementsMap,
+  originalElementsMap: ElementsMap,
+) => {
+  // keep aspect ratio for groups
+  if (property === "width") {
+    nextHeight = Math.round((nextWidth / aspectRatio) * 100) / 100;
+  } else {
+    nextWidth = Math.round(nextHeight * aspectRatio * 100) / 100;
+  }
+
+  const scale = nextHeight / initialHeight;
+
+  for (let i = 0; i < originalElements.length; i++) {
+    const origElement = originalElements[i];
+    const latestElement = latestElements[i];
+
+    resizeElementInGroup(
+      anchor[0],
+      anchor[1],
+      property,
+      scale,
+      latestElement,
+      origElement,
+      elementsMap,
+      originalElementsMap,
+    );
+  }
+};
+
+const MultiDimension = ({
+  property,
+  elements,
+  elementsMap,
+  appState,
+}: MultiDimensionProps) => {
+  const selectedGroupIds = getSelectedGroupIds(appState);
+  const groupSizes = selectedGroupIds.map((gid) => {
+    const elementsInGroup = getElementsInGroup(elements, gid);
+    const [x1, y1, x2, y2] = getCommonBounds(elementsInGroup);
+
+    return Math.round((property === "width" ? x2 - x1 : y2 - y1) * 10) / 10;
+  });
+
+  const individualSizes = elements
+    .filter((el) => !isInGroup(el) && isPropertyEditable(el, property))
+    .map((el) => Math.round(el[property] * 10) / 10);
+
+  const sizes = [...individualSizes, ...groupSizes];
+
+  const value =
+    new Set(sizes).size === 1 ? Math.round(sizes[0] * 100) / 100 : "Mixed";
+
+  const editable = sizes.length > 0;
```

Comment:
this needs to be rewritten to avoid so much wasteful memory allocation just to figure out if the values are the same or not. Given that groups are involved makes this trickier though.

I'd also consider factoring this out so that `MultiDimension`/`MultiPosition` etc. doesn't repeat the same operations.